### ISSUE TITLE
hidePoweredBy error when using server redirect

### ISSUE THREAD

**Issue by ronenteva** [Baroshem/nuxt-security#315]
### Version

nuxt-security: v1.0.0-rc.5
nuxt: v3.8.2

### Steps to reproduce
`/server/routes/go/google.js`:
```
export default defineEventHandler(event => {
  return sendRedirect(event, 'https://google.com');
});
```

### What is actually happening?
```
Error [ERR_HTTP_HEADERS_SENT]: Cannot remove headers after they are sent to the client
```

**Comment by vejja** [comment]
Hi @ronenteva
I cannot reproduce, it works on my end and I'm being redirected without errors
Can you provide more context or ideally a minimal repro ?

**Comment by ronenteva** [comment]
After looking at it again, it seems that the error won't show up unless you use a nitro plugin to catch it:

`/server/routes/go/google.js`
```
export default defineEventHandler(event => {
  return sendRedirect(event, 'https://google.com');
});
```

`/server/plugins/error.js`
```
export default defineNitroPlugin(nitroApp => {
  nitroApp.hooks.hook('error', (error, {event}) => {
    console.log(error);
  });
});
```

```
Error [ERR_HTTP_HEADERS_SENT]: Cannot remove headers after they are sent to the client
```

**Comment by vejja** [comment]
Ok thanks, I am now able to reproduce.
I'll produce a patch for this.

I think this is a bug in Nitro, as they seem to call `onBeforeResponse` after the response is sent. If confirmed I'll also file an issue there.

Thanks for reporting 💜

**Comment by vejja** [comment]
PR #318 fixes

**Comment by vejja** [comment]
Also filed related issue at https://github.com/unjs/h3/issues/596

**Comment by Baroshem** [comment]
Thanks @vejja for fixing this!

**Comment by Baroshem** [comment]
Closing as it was released with 1.0.0 :)
